macOS/UITests/FireWindowByDefaultTests.swift

```diff
@@ -0,0 +1,240 @@
+//
+//  FireWindowByDefaultTests.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import XCTest
+
+class FireWindowByDefaultTests: UITestCase {
+
+    private var dataClearingButton: XCUIElement!
+    private var generalButton: XCUIElement!
+    private var openFireWindowByDefaultToggle: XCUIElement!
+    private var reopenAllWindowsFromLastSessionRadio: XCUIElement!
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+        continueAfterFailure = false
+
+        // Assume feature flag is on by default
+        app = XCUIApplication.setUp(featureFlags: ["openFireWindowByDefault": true])
+
+        dataClearingButton = app.buttons["PreferencesSidebar.dataClearingButton"]
+        generalButton = app.buttons["PreferencesSidebar.generalButton"]
+        openFireWindowByDefaultToggle = app.checkBoxes["PreferencesDataClearingView.openFireWindowByDefault"]
+        reopenAllWindowsFromLastSessionRadio = app.radioButtons["PreferencesGeneralView.stateRestorePicker.reopenAllWindowsFromLastSession"]
+
+        app.enforceSingleWindow()
+    }
+
+    override func tearDownWithError() throws {
+        // Reset both settings to default after each test
+        resetToDefaultSettings()
+        try super.tearDownWithError()
+    }
+
+    // MARK: - Test Cases
+
+    func testFireWindowByDefaultEnabled() {
+        // Navigate to Settings -> Data Clearing
+        navigateToDataClearingSettings()
+
+        // Turn on the 'Open Fire Window by Default' setting
+        if openFireWindowByDefaultToggle.value as? Bool == false {
+            openFireWindowByDefaultToggle.click()
+        }
```

Comment:
please use `openFireWindowByDefaultToggle.toggleCheckboxIfNeeded(to: true, ensureHittable: app.ensureHittable)`